posts/2020-11-20-JNDI-gRPC.adoc

```diff
@@ -0,0 +1,361 @@
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14650[MP GraphQL does not scan JARs in WEB-INF/lib for GraphQL components]
++
+Classes annotated with things like `@GraphQLApi`, `@Type`, etc. or types referenced from root level queries that exist in JARs in the WEB-INF/lib directory were not processed. This meant that only classes in the WEB-INF/classes directory were processed by the MP GraphQL runtime. 
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14582[Prevent jsonp-1.0 and jsonpContainer-1.1 from both starting]
++
+When `jsonp-1.0` and `jsonContainer-1.1` were configured at the same time it would cause a cryptic CWWKE0702E error of:
+[source, text]
+----
+CWWKE0702E: Could not resolve module: com.ibm.websphere.javaee.jsonp.1.0 [265]
+Bundle was not resolved because of a uses constraint violation.
+----
+
+The error handling of this invalid configuration was improved by giving the user an error message of:
+[source, text]
+----
+CWWKF0033E: The singleton features com.ibm.websphere.appserver.jsonpImpl-1.1.0 and com.ibm.websphere.appserver.jsonpImpl-1.0.0 cannot be loaded at the same time. The configured features jsonpContainer-1.1 and jsonp-1.0 include one or more features that cause the conflict. Your configuration is not supported; update server.xml to remove incompatible features.
+----
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14597[Increased CPU when moving from Liberty 19.0.0.6 to newer releases]
++
+When moving from 19.0.0.6 to a newer release, there was an issue with an increase in CPU. This was caused by a fix for a previous issue which made the processing of Audit features method arguments to be performed whether the feature was enabled or not. This issue was fixed by undoing the previous change and fixing the original problem differently. For more details on Liberty's performance, see this link:https://openliberty.io/blog/2020/10/21/memory-footprint-throughput.html[post].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14743[Variables in include files not recognized after config update]
++
+This issue can cause variables defined in configuration files included by server.xml to not be recognized. We added support for using variables in include elements sometime earlier this year, and to do that we have to temporarily add new variables into the registry we use to resolve values. But in this specific scenario it affected the computation we do later to determine if any variables have changed. We would end up removing variables from the permanent variable registry. For more information take a look at our link:https://openliberty.io/docs/20.0.0.11/reference/config/server-configuration-overview.html[Server Configuration Overview].
```

Comment:
I think this may be more information than is needed here.. I would suggest something like:
We added support for using variables in include elements in 20.0.0.3. The changes required to support temporary resolution of variables during processing of configuration resulted in a bug where changes to variables may not be recognized during a configuration update. For more information.....

Reply:
Done